etc/kernel-launchers/python/scripts/launch_ipykernel.py

```diff
@@ -360,7 +365,20 @@ def get_server_request(sock):
     return request_info
 
 
-def server_listener(sock, parent_pid):
+def cancel_spark_jobs(sig, frame):
+    try:
+        __spark_context.cancelAllJobs()
```

Comment:
do we need to check if `__spark_context` is set to None?

Reply:
Yes.  I believe this was mentioned previously.